Sources/PenguinStructures/NumberOperations.swift

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extension BinaryInteger {
+  /// Returns a collection of the positive integers that are coprime with `self`.
+  ///
+  /// Definition: Two numbers are coprime if their GCD is 1.
+  public var positiveCoprimes: PositiveCoprimes<Self> { .init(self) }
+
+  /// Returns positive integers that are coprime with, and smaller than, `self`.
+  ///
+  /// - SeeAlso: `positiveCoprimes`.
+  public var smallerPositiveCoprimes: [Self] {
+    var positiveSelf = self
+    if positiveSelf < 0 {
+      positiveSelf *= -1  // Workaround for lack of `Swift.abs` on `BinaryInteger`.
+    }
+    return positiveCoprimes.prefix { $0 < positiveSelf }
+  }
+}
+
+/// The positive values that are coprime with *N*.
+///
+/// Example:
+/// ```
+/// print(Array(10.positiveCoprimes.prefix(8)))  // [1, 3, 7, 9, 11, 13, 17, 19]
+/// ```
+///
+/// Note: Although there are infinitely many positive prime numbers, `PositiveCoprimes` is bounded
+/// by the maximum representable integer in `Domain`.
```

Comment:
`BinaryInteger` doesn't have such a maximum, IIRC.

Reply:
Yes indeed. I added a comment to include that distinction.